bundler.js

```diff
@@ -79,7 +81,21 @@ export async function installBundler(bundlerVersionInput, rubygemsInputSet, lock
   const floatVersion = common.floatVersion(rubyVersion)
 
   if (bundlerVersion === 'default') {
-    if (common.isBundler2dot2Default(engine, rubyVersion)) {
+    if (systemRubyUsed) {
```

Comment:
I think we shouldn't pass `systemRubyUsed` to bundler.js, it should just work if we pass the correct `rubyVersion`.
So we'd ask the RUBY_VERSION of the system ruby early on, and use that for `rubyVersion`.
We'd just skip downloading, extracting and adding to PATH if the input is system.

Reply:
Basing it on `rubyVersion` isn't quite correct as:
* Ubuntu image doesn't ship with any Bundler
* macOS 11 image ships with a default Ruby 2.7 but with Bundler 2.4.

The logic used here for system Ruby is generic enough however that we could replace the existing `common.isBundler2dot2Default` logic here and thus avoid the `systemRubyUsed`.

Reply:
> Which Ubuntu image? IIRC `ubuntu-22.04` ships with some Ruby version in PATH by default from the toolcache, and that has Bundler.

This is what I get with `ubuntu-22.04`: https://github.com/Bo98/workflow-test/actions/runs/7118996527/job/19383150052

macOS and Ubuntu runners do not use a toolcache Ruby by default. They use ones from system package managers.